File: src/main/java/com/quantumjockey/melya/controls/initialization/ButtonInitializer.java
package com.quantumjockey.melya.controls.initialization;

import javafx.scene.control.Button;
import javafx.scene.control.Tooltip;
import javafx.scene.image.Image;
import javafx.scene.image.ImageView;

public class ButtonInitializer {

    /////////// Fields //////////////////////////////////////////////////////////////////////

    private Button button;

    /////////// Constructors ////////////////////////////////////////////////////////////////

    public ButtonInitializer(Button button){
        this.button = button;
    }

    /////////// Public Methods //////////////////////////////////////////////////////////////

    public void setSkin(Image imageResource){
        this.button.setGraphic(new ImageView(imageResource));
    }

    public void setTooltip(String tooltip){
        this.button.setTooltip(new Tooltip(tooltip));
    }
}

File: src/main/scala/com/quantumjockey/melya/controls/initialization/ButtonInitializer.scala
package com.quantumjockey.melya.controls.initialization

import javafx.scene.control.{Tooltip, Button}
import javafx.scene.image.{ImageView, Image}

class ButtonInitializer(button: Button) {

  // Public Methods

  def setSkin(imageResource: Image): Unit = this.button.setGraphic(new ImageView(imageResource))

  def setTooltip(tooltip: String): Unit = this.button.setTooltip(new Tooltip(tooltip))

}
